### PR TITLE
ci: Add CodeQL analysis GitHub workflow

### PR DESCRIPTION
**Pull request by WillDaSilva**

### CODE REVIEW

**Comment by tayloramurphy** [comment]
@WillDaSilva do you actually need a review from all of the people currently assigned? I don't have context on what this PR is doing.

**Comment by WillDaSilva** [comment]
> do you actually need a review from all of the people currently assigned? I don't have context on what this PR is doing.

@tayloramurphy Nope. All reviewers were auto-assigned as code owners.

**Review by edgarrmondragon — APPROVED** [review]

**Pull request merged**
